lib/mayaUsd/ufe/MayaUsdContextOps.cpp

```diff
@@ -236,12 +236,89 @@ makeUSDReferenceFilePathRelativeIfRequested(const std::string& filePath, const U
+        // Populate list of known renderers (first menu level).
+        for (auto it = renderersAndMaterials.begin(), end = renderersAndMaterials.end(); it != end;
+             it = renderersAndMaterials.upper_bound(it->first)) {
+            items.emplace_back(it->first, it->first, Ufe::ContextItem::kHasChildren);
+        }
+    } else if (itemPath.size() == 2u) {
+        // Populate list of materials for a given renderer (second menu level).
+        const auto range = renderersAndMaterials.equal_range(itemPath[1]);
+        for (auto it = range.first; it != range.second; ++it) {
+            MStringArray materialAndIdentifier;
+            // Expects a string in the format "Material Name|MaterialIdentifer".
+            MStatus status = it->second.split('|', materialAndIdentifier);
+            if (status == MS::kSuccess && materialAndIdentifier.length() == 2) {
+                items.emplace_back(
+                    materialAndIdentifier[1].asChar(), materialAndIdentifier[0].asChar());
+            }
+        }
+    }
+}
+
+void assignExistingMaterialItems(
+    const UsdSceneItem::Ptr&         item,
+    const Ufe::ContextOps::ItemPath& itemPath,
+    Ufe::ContextOps::Items&          items)
+{
+    std::multimap<std::string, MString> pathsAndMaterials;
+    MStringArray                        materials;
+    MString                             script;
+    script.format(
+        "mayaUsdGetMaterialsInStage \"^1s\"", Ufe::PathString::string(item->path()).c_str());
+    MGlobal::executeCommand(script, materials);
+
+    for (const auto& material : materials) {
+        MStringArray pathAndMaterial;
+        // Expects a string in the format "/path1/path2/Material".
+        const int lastSlash = material.rindex('/');
+        if (lastSlash >= 0) {
+            MString pathToMaterial = material.substring(0, lastSlash);
+            pathsAndMaterials.emplace(std::string(pathToMaterial.asChar()), material);
+        }
+    }
+
+    if (itemPath.size() == 1u) {
+        // Populate list of paths to materials (first menu level).
+        for (auto it = pathsAndMaterials.begin(), end = pathsAndMaterials.end(); it != end;
+             it = pathsAndMaterials.upper_bound(it->first)) {
+            items.emplace_back(it->first, it->first, Ufe::ContextItem::kHasChildren);
+        }
+    } else if (itemPath.size() == 2u) {
+        // Populate list of to materials for given path (second  menu level).
+        const auto range = pathsAndMaterials.equal_range(itemPath[1]);
+        for (auto it = range.first; it != range.second; ++it) {
+            const int lastSlash = it->second.rindex('/');
+            if (lastSlash >= 0) {
+                MString materialName = it->second.substring(lastSlash + 1, it->second.length() - 1);
+                items.emplace_back(it->second.asChar(), materialName.asChar());
+            }
+        }
+    }
+}
+#endif
+
+inline bool sceneItemSupportsShading(const Ufe::SceneItem::Ptr& sceneItem)
```

Comment:
This one was already here, I just made it inline and removed if (now just single return line).